File: cpp/206_reverse_list.cpp
ListNode* reverseList(ListNode* head) {
        stack<int> st;
        ListNode *ptr = head;
        while(ptr){
            st.push(ptr->val);
            ptr = ptr->next;
        }
        
        ptr = head;
        while(ptr){
            ptr->val = st.top();
            st.pop();
            ptr=ptr->next;
        }
        
        return head;
        
    }

File: java/206_reverse_list.java
public ListNode reverseList(ListNode head) {
        if(head==null)
            return null;
        
        Stack<Integer> st = new Stack();
        ListNode ptr = head;
        while(ptr!=null){
            st.push(ptr.val);
            ptr = ptr.next;
        }
        ptr = head;
        while(ptr!=null){
            ptr.val = st.pop();
            ptr = ptr.next;
        }
        
        return head;
    }